### PR TITLE
Add the argo mercator example

### PR DESCRIPTION
**Pull request by stefraynaud**
It make a simple comparison between an ARGO profile and Mercator outputs

### CODE REVIEW

File: examples/plot_mercator_argo.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+# %%
+# Here we select the last ARGO profile as a reference profile
+# since it better samples de mediterranean water vein.
+ds_argo_prof = ds_argo.isel(time=-1).squeeze(drop=True)
+
+# %%
+# Mercator
+# ~~~~~~~~
+#
+# The Mercator data come from the IBI configuration and were download
```
Comment (vquilfen) [inline examples/plot_mercator_argo.py:111]: maybe add the CMEMS product reference or web link

File: examples/plot_mercator_argo.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+# from the CMEMS site and undersampled by a factor 3 in longitude and latitude.
+ds_merc = xoa.open_data_sample("ibi-argo-7900573.nc").xoa.decode()
+ds_merc = ds_merc.isel(depth=slice(None, None, -1))
+ds_merc = ds_merc.assign_coords(depth=-ds_merc.depth)
+print(ds_merc)
+
+# %%
+# This block of data can be viewed as a ensemble of profiles that emulates
```
Comment (vquilfen) [inline examples/plot_mercator_argo.py:119]: a set of profiles ?

File: examples/plot_mercator_argo.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+plt.legend()
+
+# %%
+# We used the :func:`xoa.geo.get_extent` to compute a square geographical extent based
+# on the coordinates of a dataset with an added margin.
+
+# %%
+# Now interpolate Mercator onto ARGO time and position for both the ensemble and
+# the reference profile. All interpolations can be performed
+# with the :meth:`~xarray.Dataset.interp` method since with dealing with
```
Comment (vquilfen) [inline examples/plot_mercator_argo.py:167]: un 'with' en trop devant 'dealing' ?

File: examples/plot_mercator_argo.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+plt.legend()
+
+# %%
+# Therefore, if we accept an uncertainty in the positioning of the ocean
+# structure in the model, it differences with (assimilated) observations
+# become acceptables.
+
+# %%
+# As you can see on the previous figures, the ARGO profiles come with
+# a very high vertical resolution at all depth, whereas the model
```
Comment (vquilfen) [inline examples/plot_mercator_argo.py:195]: depths

File: examples/plot_mercator_argo.py [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+# that supports the linear (:attr:`~xoa.regrid.regrid1d_methods.linear` method) interpolation
+# to higher resolutions and the cell averaging
+# (:attr:`~xoa.regrid.regrid1d_methods.cellave` method) to lower resolutions.
+sal_merc_ens_o = xregrid.regrid1d(ds_merc_ens.sal, ds_argo_prof.depth, method="linear")
+sal_merc_prof_o = xregrid.regrid1d(ds_merc_prof.sal, ds_argo_prof.depth, method="linear")
+sal_argo_prof_m = xregrid.regrid1d(ds_argo_prof.sal, ds_merc_prof.depth, method="cellave")
+merc_prof_edges = xgrid.get_edges(ds_merc_prof.depth, "depth")
+
+# %%
+# Let's plots them.
```
Comment (vquilfen) [inline examples/plot_mercator_argo.py:212]: plot

**Review by vquilfen — APPROVED** [review]

**Review by mdussa — APPROVED** [review]

File: examples/plot_mercator_argo.py (hunk 2 above)
Comment (stefraynaud) [reply examples/plot_mercator_argo.py:119]: "ensemble" refers to the ensemble data assimilation in which the "set" emulates the uncertainties.

**Pull request merged**